packages/dd-trace/src/plugins/util/ip_extractor.js

```diff
@@ -41,56 +41,50 @@ function extractIp (config, req) {
   const headers = req.headers
   if (config.clientIpHeader) {
     if (!headers) return
-    const header = headers[config.clientIpHeader]
-    if (!header) return
 
-    return findFirstIp(header)
+    const ip = findFirstIp(headers[config.clientIpHeader])
+    return ip.public || ip.private
   }
 
-  const foundHeaders = []
+  let firstPrivateIp
   if (headers) {
     for (let i = 0; i < ipHeaderList.length; i++) {
-      if (headers[ipHeaderList[i]]) {
-        foundHeaders.push(ipHeaderList[i])
+      const firstIp = findFirstIp(headers[ipHeaderList[i]])
+
+      if (firstIp.public) {
+        return firstIp.public
+      } else if (!firstPrivateIp && firstIp.private) {
+        firstPrivateIp = firstIp.private
       }
     }
   }
 
-  if (foundHeaders.length === 1) {
-    const header = headers[foundHeaders[0]]
-    const firstIp = findFirstIp(header)
-
-    if (firstIp) return firstIp
-  } else if (foundHeaders.length > 1) {
-    log.error(`Cannot find client IP: multiple IP headers detected ${foundHeaders}`)
-    return
-  }
-
-  return req.socket && req.socket.remoteAddress
+  return firstPrivateIp || (req.socket && req.socket.remoteAddress)
 }
 
 function findFirstIp (str) {
   let firstPrivateIp
```

Comment:
```suggestion
  const result = {}
  if (!str) return result
  
  // ...
  
  if (!result.private) result.private = chunk
  
  // ...
  
  return result
```

Reply:
way more readable imo

Reply:
Then maybe assign `result.public = chunk ` when public is found and `break`? xDD

```suggestion
  const result = {}
  if (!str) return result

  const splitted = str.split(',')

  for (let i = 0; i < splitted.length; i++) {
    const chunk = splitted[i].trim()

    // TODO: strip port and interface data ?

    const type = net.isIP(chunk)
    if (!type) continue

    if (!privateIPMatcher.check(chunk, type === 6 ? 'ipv6' : 'ipv4')) {
      // it's public, return it immediately
      result.public = chunk
      break
    }

    // it's private, only save the first one found
    if (!result.private) result.private = chunk
  }

  return result
```

Reply:
LGTM